atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2123,7 +2132,7 @@ protected ConflictHandler getConflictHandlerForTable(TableReference tableRef) {
     /**
      * Make sure we have all the rows we are checking already locked before calling this.
      */
-    protected void throwIfConflictOnCommit(LockToken commitLocksToken, TransactionService transactionService)
+    protected void throwIfConflictOnCommit(Set<LockToken> commitLocksToken, TransactionService transactionService)
```

Comment:
Technically, these could be sourced from the manager? I'll need to think what's best for reading this code and understanding hidden dependencies.

Reply:
It already chooses only certain state to be passed in explicitly vs implicitly (so locks get passed explicitly, but the writes are implicit), so I don't think it would be a big deal

Reply:
Ya, this minimizes the diff, so lgtm.